newIDE/app/src/UI/SemiControlledAutoComplete.js

```diff
@@ -137,219 +70,182 @@ type State = {|
   };
 
-  /**
-   * Allow to override the value being written in the autocomplete, if any.
-   * Usually you don't want to do that as the point of having a "SemiControlled" auto-complete
-   * is that the value being written by the user is the source of truth.
-   *
-   * In some cases, you want to force a new value when the auto-complete is focused (for example,
-   * if you opened a native dialog selector that is going to make the auto-complete focused/blurred/
-   * focused again, in a manner that depends on the OS).
-   * Call this function to do so.
-   *
-   * @param newValue The new value to set in the auto complete. If the field is not being focused,
-   * nothing will be forced (the value props will be used when the field is focused).
-   */
-  forceInputValueTo(newValue: string) {
+  focus() {
+    if (this._input.current) this._input.current.focus();
+  }
+
+  forceInputValueTo(newValue: string): void {
     if (this.state.inputValue !== null) {
       this.setState({
         inputValue: newValue,
       });
     }
   }
 
-  focus() {
-    if (this._input.current) this._input.current.focus();
-  }
+  isOptionDisabled = (option: Option): boolean => {
+    if (option.type === 'separator') return true;
+    return false;
+  };
 
   render() {
-    const { props, state } = this;
-    const currentInputValue =
-      state.inputValue !== null ? state.inputValue : props.value;
+    const getCurrentInputValue = (): string =>
+      this.state.inputValue !== null ? this.state.inputValue : this.props.value;
+
+    const filterFunction = (options: DataSource, state: Object): DataSource => {
+      const lowercaseInputValue = getCurrentInputValue().toLowerCase();
+      const optionList = options.filter(option => {
+        if (option.type === 'separator') return true;
+        if (!option.text) return true;
+        return option.text.toLowerCase().indexOf(lowercaseInputValue) !== -1;
+      });
 
-    const helperText = props.helperMarkdownText ? (
-      <MarkdownText source={props.helperMarkdownText} />
+      if (
+        !optionList.filter(option => option.type !== 'separator' && option.text)
+          .length
+      )
+        return [];
+
+      // Remove divider(s) if they are at the start or end of array
+      while (
+        optionList[optionList.length - 1] !== undefined &&
+        optionList[optionList.length - 1].type !== undefined
+      )
+        optionList.pop();
+      while (
+        optionList[optionList.length - 1] !== undefined &&
+        optionList[0].type !== undefined
+      )
+        optionList.shift();
+
+      return optionList;
+    };
+
+    const helperText = this.props.helperMarkdownText ? (
+      <MarkdownText source={this.props.helperMarkdownText} />
     ) : null;
 
+    const getOptionLabel = (option: Option): string => {
+      if (option.value) return option.value;
+      return getCurrentInputValue();
+    };
+
+    const handleChange = (event, option: Option | string): void => {
```

Comment:
Add a type for event, even if you have to write it yourself it's better than nothing:
`{ key: string }`

because there will always be an event right?
Then remove the `typeof event.key === 'string'`

Reply:
Ping on this: do you know the type of the event? :)